### PR TITLE
docs: updating v3->v4 migration docs with fee middleware

### PR DESCRIPTION
**Pull request by seantking**
## Description

closes: #2089 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/migrations/v3-to-v4.md [hunk 1]
```diff
@@ -125,6 +131,8 @@ if err := k.icaControllerKeeper.RegisterInterchainAccount(ctx, msg.ConnectionId,
 }
 ```
 
+In order to incentivize ICS27 packets using the fee middleware module a middleware stack that contains the fee middleware must be configured. Take a look at the following diff for an [example setup](https://github.com/cosmos/ibc-go/pull/1432/files#diff-d18972debee5e64f16e40807b2ae112ddbe609504a93ea5e1c80a5d489c3a08aL366).
```
Comment (colin-axner) [inline docs/migrations/v3-to-v4.md:134]: should this be under the chains section?

File: docs/migrations/v3-to-v4.md [hunk 2]
```diff
@@ -18,6 +18,12 @@ No genesis or in-place migrations required when upgrading from v1 or v2 of ibc-g
 
 ## Chains
 
+### Fee Middleware
+
+The Fee Middleware module, as the name suggests, plays the role of an IBC middleware and as such must be configured by chain developers to route and handle IBC messages correctly.
+
+Please read the Fee Middleware [integration documentation](https://ibc.cosmos.network/main/middleware/ics29-fee/integration.html) for an in depth guide on how to congfigure the module correctly in order to incentivize IBC packets.
```
Comment (colin-axner) [inline docs/migrations/v3-to-v4.md:25]: This looks good, but I don't know if we have any docs on actually integrating ics29 as an SDK module. Maybe we should just update the docs?

**Review by colin-axner — COMMENTED** [review]
LGTM, but I think we are still missing a bit of information on integrating ics29. If a chain goes from v3, we should be linking enough information to upgrade without looking at our code. Maybe we should do this in a followup pr?

File: docs/migrations/v3-to-v4.md (hunk 2 above)
Comment (seantking) [reply docs/migrations/v3-to-v4.md:25]: I can make a follow up PR to update the docs as well?

**Review by colin-axner — APPROVED** [review]

**Comment by colin-axner** [comment]
opened #2354

**Review by damiannolan — APPROVED** [review]

**Pull request merged**